taglib/toolkit/tfilestream.cpp

```diff
@@ -158,6 +168,29 @@ FileStream::FileStream(FileName fileName, bool openReadOnly)
   }
 }
 
+FileStream::FileStream(int fileDescriptor, bool openReadOnly)
+  : d(new FileStreamPrivate(""))
+{
+  // First try with read / write mode, if that fails, fall back to read only.
+
+  if(!openReadOnly)
+    d->file = openFile(fileDescriptor, false);
+
+  if(d->file != InvalidFileHandle)
+    d->readOnly = false;
+  else
+    d->file = openFile(fileDescriptor, true);
+
+  if(d->file == InvalidFileHandle)
+  {
+# ifdef _WIN32
+    debug("Could not open file using file descriptor");
+# else
+    debug("Could not open file file using file descriptor");
```

Comment:
`file` is repeated twice. Why does this section have an `#ifdef`?

Reply:
Thank you. I've fixed it.